chall05/vde-dios.c

```diff
@@ -0,0 +1,31 @@
+char   ft_lower(char letter)
+{
+    if (letter >= 'A' && letter <= 'Z')
+        return (letter + 32);
+    return (letter);
+}
+
+int ft_ie_except_after_c(char* str)
+{
+    char *p;
+
+    if (str)
+    {
+        p = str;
+        while (*str)
+        {
+            if (ft_lower(*str) == 'e')
+            {
+                if (str != p && (str - 1)!= p)
+                    if (ft_lower(*(str - 1)) == 'i' && ft_lower(*(str - 2)) == 'c')
+                        return (0);
+                if (*(str + 1))
+                    if (ft_lower(*(str + 1)) == 'i' && (str == p || ft_lower(*(str - 1)) != 'c'))
+                        return (0);
+            }
+            str++;
```

Comment:
Having to use your `ft_lower` function prevents you from doing so, but you could have used `strstr` to look for the wanted occurrences. For example you know that you should never find `cie`, this could be solved in an `if`.  
Later you would still have to loop to check for `ei` that are not preceded with `c`.

Reply:
That is true, however I tried to resolve the problem without using external functions since the problem seemed simple enough for making it with simple conditionals. I do not know if using strstr would lead to a performance improvement, rather than having a more readable code (that is true)